packages/backend/src/errors/step.ts

```diff
@@ -0,0 +1,12 @@
+import { IStepError } from '@plumber/types'
+
+export default class StepError extends Error {
+  details = {}
+  constructor(error: IStepError) {
```

Comment:
nit: thinking if we we can use es2022's `ErrorOptions` and support storing the underlying error in the `cause` field.

https://nodejs.org/docs/latest-v18.x/api/errors.html#new-errormessage-options

Just need to do a few this to set this up:
1. Add `es2022.error` to `lib` in tsconfig
2. Be careful that we're not logging raw axios errors (which may have errors - this should be handled by `HttpError` and our interceptors though, so not too worried there)

Reply:
ok sounds like a good idea thks